### PR TITLE
[PSM Interop] Return trailing metadata.

### PR DESCRIPTION
**Pull request by eugeneo**
1. Trailing metadata is now reported
2. messages.proto was synchronized.
3. Corrected order of arguments in EXPECT_EQ so the output makes sense now.

### CODE REVIEW

File: src/proto/grpc/testing/messages.proto [hunk 1]
```diff
@@ -309,3 +317,23 @@ message TestOrcaReport {
   map<string, double> request_cost = 3;
   map<string, double> utilization = 4;
 }
+
+enum HookRequestCommand {
```
Comment (yashykt) [inline src/proto/grpc/testing/messages.proto:321]: I don't see these being used

Comment (eugeneo) [reply src/proto/grpc/testing/messages.proto:321]: They are from that other PR. This file source of truth is https://github.com/grpc/grpc-proto/blob/master/grpc/testing/messages.proto that already has the messages.

Comment (yashykt) [reply src/proto/grpc/testing/messages.proto:321]: can you merge master then please?

Comment (eugeneo) [reply src/proto/grpc/testing/messages.proto]: Rebased on top of "hook server" change, should be easier to follow now.

**Review by yashykt — APPROVED** [review]

**Pull request merged**